rules/hmap.bzl

```diff
@@ -78,22 +78,12 @@ def _make_headermap_impl(ctx):
     cc_info_provider = CcInfo(
         compilation_context = compilation_context,
     )
-    swift_info_provider = swift_common.create_swift_info(
-        modules = [swift_common.create_module(
-            name = ctx.attr.name,
-            clang = swift_common.create_clang_module(
-                compilation_context = compilation_context,
-                module_map = None,
-            ),
-        )],
-    )
     providers = [
         DefaultInfo(
             files = depset([headermap]),
         ),
         apple_common.new_objc_provider(),
```

Comment:
We should try to remove this as well

Reply:
my understanding is bazel 6 still need it. but i can be wrong. 
I feel that demands a separate PR to just remove all `apple_common.objc` references? Also helps when bump to bazel 7.4 / nightly as the API is just gone now: https://bazel.build/rules/lib/toplevel/apple_common